resources/measures/upgrade_hvac_pump/measure.rb

```diff
@@ -0,0 +1,861 @@
+      end
+    pump_var_part_load_curve_coeff2_weighted_avg =
+      if pump_rated_flow_total > 0.0
+        pump_var_part_load_curve_coeff2_weighted_sum / pump_rated_flow_total
+      else
+        0.0
+      end
+    pump_var_part_load_curve_coeff3_weighted_avg =
+      if pump_rated_flow_total > 0.0
+        pump_var_part_load_curve_coeff3_weighted_sum / pump_rated_flow_total
+      else
+        0.0
+      end
+    pump_var_part_load_curve_coeff4_weighted_avg =
+      if pump_rated_flow_total > 0.0
+        pump_var_part_load_curve_coeff4_weighted_sum / pump_rated_flow_total
+      else
+        0.0
+      end
+
+    [
+      applicable_pumps,
+      pump_rated_flow_total,
+      pump_motor_eff_weighted_average,
+      pump_motor_bhp_weighted_average,
+      pump_var_part_load_curve_coeff1_weighted_avg,
+      pump_var_part_load_curve_coeff2_weighted_avg,
+      pump_var_part_load_curve_coeff3_weighted_avg,
+      pump_var_part_load_curve_coeff4_weighted_avg
+    ]
+  end
+
+  # get motor efficiency from nominal power
+  def self.estimate_motor_efficiency_pcnt(nominal_power_w)
+    nominal_power_kw = nominal_power_w / 1000.0
+
+    # Regression parameters from Python popt_fixed
+    a = 1.64644705
+    b = 92.25875553
+    c = 50.22494607
+    d = 0.00061996
+
+    # Fixed breakpoint
+    x0 = 5.0
+
+    # Efficiency bounds (%)
+    eff_min = 90.53
+    eff_max = 95.95
+
+    if nominal_power_kw <= 0
+      raise ArgumentError, 'Nominal power must be greater than 0'
+    end
+
+    if nominal_power_kw < x0
+      motor_efficiency_pcnt = (a * Math.log(nominal_power_kw)) + b
+    else
+      # Compute e to ensure continuity at x0
+      left_val = (a * Math.log(x0)) + b
+      right_val = c * (1 - Math.exp(-d * x0))
+      e = left_val - right_val
+
+      motor_efficiency_pcnt = (c * (1 - Math.exp(-d * nominal_power_kw))) + e
+    end
+
+    # Clip output to [90.53%, 95.95%]
+    motor_efficiency_pcnt.clamp(eff_min, eff_max)
+  end
+
+  # get part-load fraction of full load power curve
+  def self.curve_fraction_of_full_load_power(model)
+    # Define coefficients for the cubic curve
+    coeff_a = 0.0
+    coeff_b = 0.0
+    coeff_c = 0.1055
+    coeff_d = 0.8945
+    # Define a cubic curve with example coefficients
+    curve = OpenStudio::Model::CurveCubic.new(model)
+    curve.setName('Fraction of Full Load Power Curve')
+    curve.setCoefficient1Constant(coeff_a) # y-intercept
+    curve.setCoefficient2x(coeff_b) # linear term
+    curve.setCoefficient3xPOW2(coeff_c) # quadratic term
+    curve.setCoefficient4xPOW3(coeff_d) # cubic term
+    curve.setMinimumValueofx(0.0)
+    curve.setMaximumValueofx(1.0)
+    curve.setMinimumCurveOutput(0.0)
+    curve.setMaximumCurveOutput(1.0)
+    return curve, coeff_a, coeff_b, coeff_c, coeff_d
+  end
+
+  # iteratively compute the design motor power based on flow, head, and pump efficiency.
+  def self.compute_design_motor_power(flow_m3_per_s, head_pa, eta_pump = 0.75, tolerance = 0.001,
```

Comment:
this method is trying to estimate motor efficiency based on manufacturer data. it actually uses iterative approach since motor eff is correlated with pump power from manufacturer data. measure doc has more details on this.